### PR TITLE
Remove Email-config section

### PR DESCRIPTION
**Pull request by AllForNothing**
1. Remove the Email-config section, as the related function has been removed in Harbor https://github.com/goharbor/harbor/issues/17683

Signed-off-by: AllForNothing <sshijun@vmware.com>

### CODE REVIEW

File: docs/administration/general-settings/_index.md [hunk 1]
```diff
@@ -3,13 +3,7 @@ title: Configure Global Settings
 weight: 20
 ---
 
-You can configure Harbor to connect to an email server, set the registry in read-only mode, and configure Harbor so that only system administrators can create projects.
-
-## Configure an Email Server
-
-You can configure Harbor to connect to an email server. The email server is only used to send out responses to users who request to reset their password.
-
```
Comment (OrlinVasilev) [inline docs/administration/general-settings/_index.md:11]: is that deprecated ?

File: docs/administration/managing-users/create-users-db.md [hunk 2]
```diff
@@ -18,4 +18,4 @@ In database authentication mode, the Harbor system administrator creates user ac
     - The email address is used for password recovery
     - The password must contain at least 8 characters with 1 lowercase letter, 1 uppercase letter and 1 numeric character
 
-If users forget their password, there is a **Forgot Password** in the Harbor log in page. To use this feature, you must [configure an email server](../general-settings/_index.md).
```
Comment (OrlinVasilev) [inline docs/administration/managing-users/create-users-db.md:21]: is that gone?

Comment (AllForNothing) [reply docs/administration/managing-users/create-users-db.md:21]: Yes. Password reset function has been remote in 2.3, by  https://github.com/goharbor/harbor/pull/14829.

**Review by OrlinVasilev — APPROVED** [review]
lgtm

**Review by wy65701436 — APPROVED** [review]
lgtm

**Pull request merged**